server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/restore/RestoreSnapshotResponse.java

```diff
@@ -86,4 +88,56 @@ public XContentBuilder toXContent(XContentBuilder builder, ToXContent.Params par
         builder.endObject();
         return builder;
     }
+
+    public static RestoreSnapshotResponse fromXContent(XContentParser parser) throws IOException {
+        RestoreSnapshotResponse response = new RestoreSnapshotResponse();
```

Comment:
@hub-cap correct; I implemented this identically to `CreateSnapshotResponse`. Similarly to `RestoreInfo` there is a mismatch between the object structure and the `XContent` structure. For `RestoreInfo` I managed to use `ObjectParser` but I had a hard time figuring out how to make it work. I also agree that if we change it here, we should change it in `CreateSnapshotResponse` as well.

Reply:
This definitely does not ignore unknown fields. It will fail if snapshot or accepted is not in the body. @nik9000 how should we handle this? We can go back to the older for loop style of these, or just fix it here using object parser. Id prefer the latter given @danielmitterdorfer has infinite time (which i dont know).

Reply:
Failing on less stuff is fine, but we should indeed ignore unknown fields. I think `ConstructingObjectParser` would work fine for this if you declare `shapshot` as an `optionalConstructorArg` and you declare `accepted` as boolean `optionalConstructorArg` and throw it on the floor when building the actual object. It isn't *super* efficient but it will properly handle unknown fields and it isn't *too* bad.

Reply:
And it means we don't need to think too hard about the parsing. Which I don't think is worth it for an API as low traffic as this.